crates/edr_eth/src/signature.rs

```diff
@@ -52,14 +51,14 @@ pub enum SignatureError {
 /// A fakeable signature which can either be a fake signature or a real ECDSA
 /// signature.
 #[derive(Clone, Debug)]
-pub struct Fakeable<SignatureT: Recoverable + Signature> {
+pub struct Fakeable<SignatureT: Signature> {
```

Comment:
Previously I lazily recovered the address of a signature. Now I always ensure that the address is either provided or recovered. This ensures that we get the signature's recovery error upon construction, when it's clearest to the user  that they made a mistake.

The performance impact should be minimal, as we almost always know the `Address` anyway or were validating that the signature was valid already.